src/main/scala/strawman/collection/Map.scala

```diff
@@ -0,0 +1,44 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Option
+import scala.annotation.unchecked.uncheckedVariance
+import scala.Predef.???
+
+/** Base Map type */
+trait Map[K, +V]
+  extends Iterable[(K, V)]
```

Comment:
*Only* if there's actually a `Tuple2` behind it. But not otherwise:
```scala
scala> case class Foo(_1: Int, _2: Int) extends Product2[Int, Int]
defined class Foo

scala> val xs: Iterable[Product2[Int, Int]] = List(Foo(1, 2))
xs: Iterable[Product2[Int,Int]] = List(Foo(1,2))

scala> for ((key, value) <- xs) println(s"$key: $value")

```
That's right: it *does not even crash*! It filters out things that are not `Tuple2`s.